empress/recon_vis/recon_viewer.py

```diff
@@ -96,55 +237,351 @@ def render_parasite_helper(fig, node, recon, host_lookup, show_internal_labels,
+    :return A tuple consisting of the left node and right node
+    """
+    mapping_node = recon.mapping_of(node.name)
+    event = recon.event_of(mapping_node)
+    left_mapping_node = event.left
+    right_mapping_node = event.right
+    left_node_name = left_mapping_node.parasite
+    right_node_name = right_mapping_node.parasite
+
+    left_node = parasite_lookup[left_node_name]
+    right_node = parasite_lookup[right_node_name]
+
+    return left_node, right_node
+
+
+def render_curved_line_to(node_pos: plot_tools.Position, other_pos: plot_tools.Position, fig: plot_tools.FigureWrapper):
+    """
+    Renders a curved line from one point to another
+    :param node_pos: x and y position of a node
+    :param other_pos: x and y position of another node
+    :param fig: Figure object that visualizes trees using MatplotLib
+    """
+    mid_pos = plot_tools.Position(node_pos.x, other_pos.y)
+    fig.line(node_pos, mid_pos, PARASITE_EDGE_COLOR)
+    fig.line(mid_pos, other_pos, PARASITE_EDGE_COLOR)
+
+
+def render_transfer_branch(node_pos: plot_tools.Position, right_pos: plot_tools.Position, fig: plot_tools.FigureWrapper,  node: tree.Node, host_lookup: dict, recon: recon.Reconciliation, right_node: tree.Node):
+    """
+    Renders a transfer branch
+    :param node_xy: x and y position of a node
+    :param right_pos: x and y position of the right child of a node
+    :param fig: Figure object that visualizes trees using MatplotLib
+    :param node: Node object
+    :param host_lookup: Dictionary with host node names as the key and host node objects as the values
+    :param recon: Reconciliation object
+    :param right_node: The right node object of node
+    """
+    mapping_node = recon.mapping_of(node.name)
+    host_node = host_lookup[mapping_node.host]
+
+    child_mapping_node = recon.mapping_of(right_node.name)
+    child_host_node = host_lookup[child_mapping_node.host]
+
+    # Check temporal consistency of transfer event
+    if child_host_node.parent_node.layout.col < node.layout.col:
+        # Draw right node, which is transfered
+        mid_pos = plot_tools.Position(node_pos.x, right_pos.y)          #xy coords of midpoint
+        y_midpoint = abs(mid_pos.y + node_pos.y) / 2         #value of midpoint between mid_xy and parent node
+
+        # Determine if transfer is upwards or downwards, and draw triangle accordingly
+        is_upwards = True if y_midpoint < mid_pos.y else False
+        if is_upwards:
+            fig.up_triangle((node_pos.x, y_midpoint), PARASITE_EDGE_COLOR)
+        else:
+            fig.down_triangle((node_pos.x, y_midpoint), PARASITE_EDGE_COLOR)
+
+        # Draw branch to midpoint, then draw branch to child
+        fig.line(node_pos, mid_pos, PARASITE_EDGE_COLOR)
+        fig.line(mid_pos, right_pos, PARASITE_EDGE_COLOR)
+    else:
+        transfer_edge_color = (PARASITE_EDGE_COLOR[0] , PARASITE_EDGE_COLOR[1] , PARASITE_EDGE_COLOR[2], TRANSFER_TRANSPARENCY)
```

Comment:
You might want to create a new helper function in `plot_tools`:
```python
from typing import Tuple
...
def transparent_color(col: Tuple[int, int, int, float], alpha: float):
    return col[0:3] + (alpha,)
```
If you don't like the name `transparent_color`, you could go for something like `new_transparent_color` or `color_with_transparency`. Try to avoid names like `set_color_transparency` because you are not modifying the transparency of the original color.

Reply:
Done